balancer/grpclb/grpclb_remote_balancer.go

```diff
@@ -260,10 +260,11 @@ func (lb *lbBalancer) newRemoteBalancerCCWrapper() error {
 	// The grpclb server addresses will set field ServerName, and creds will
 	// receive ServerName as authority.
 	target := lb.manualResolver.Scheme() + ":///grpclb.subClientConn"
-	cc, err := grpc.Dial(target, dopts...)
+	cc, err := grpc.NewClient(target, dopts...)
```

Comment:
@dfawley do you know if this non-test usage can be effected by #7556? If so, should we hold-off on making this change? 

Similar question for the following files:
* balancer/rls/control_channel.go
* credentials/alts/internal/handshaker/service/service.go